### DISCUSSION TITLE
"Møtereferat" fra joggetur med Matre og Markus

### DISCUSSION THREAD

**Discussion by fyksen** [TreshLaps/Common#8] [General]
* Markus skal ha møte med forskningsgruppa fredag, kommer da tilbake med eksakt hvilke data de vil ha. Han skal også se på [strava dokumentasjonen](https://developers.strava.com/docs/reference/), for å se hvilke data vi burde polle for å oppnå dette.
* Vi må enes om hvilke format vi skal ha for å hente ut laktat på øktene, så folk kan så smått begynne å fylle ut øktene sine riktig. Mitt forslag:
```
(💉/🩸)$laktatverdi etter $LAP
```

Hvor LAP er hvilket terskeldrag målingen ble tatt på. (Husk at antall totale laps er høyere enn terskeldrag, da pausene også er en lap). Tenker også det er litt hyggelig at folk kan velge selv hvilke emoji de vil bruke, av 🩸 og 💉.
 
f.eks:

```
💉2,5 etter 4.
🩸3,4 etter 7.
```

* Et mål er at det skal være så lite input fra brukeren som mulig. Matre og Markus syntes derfor det var greiest at man ikke måtte merke øktene som ```workout``` for at de skal dukke opp som terskeldrag i systemet.

## Definere terskel
En av de store avgjørelsene er hvordan vi skal definere hva slags intensitet et drag er på. Siden det er varierende nivå på de som skal bruke programmet, kan vi ikke sette statiske variabler. Det er heller ikke optimalt å få brukeren til å sette de selv, da det er vanskelig å huske bakover i tid hva terskelfarta var for noen år tilbake i tid.

En mulighet Markus snakka *mye* om, var å "clustre" lapene, så forsøke å definere tre grupper ut fra denne dataen. (1. roligtur, 2. terskel, 3. over terskel). Vi kan dele opp datasettet i pr måned. På den måten slipper vi input fra brukeren for å definere dette.
Jeg er ikke veldig kjent med clustering av data, men ser ut som det kan være noe dere programmere har [kjennskap](https://www.analyticsvidhya.com/blog/2016/11/an-introduction-to-clustering-and-different-methods-of-clustering) til?

**Comment by mwarvik** [comment]
Jeg tenker det er fint å støtte flere formater på laktatmåling, dersom man bruker emojien kan man greit parse mange forskjellige måter å skrive det på. Og tenker det er enklere å si at man skriver lap# for hvilket intervalldrag man er på. Ellers må man begynne å regne ut hvor mange runder oppvarming etc man har.

**Reply by fyksen** [reply]
Jepps, det var det jeg mente med det jeg skrev der. "Hvor LAP er hvilket terskeldrag målingen ble tatt etter". Skulle det stått.

**Reply by mwarvik** [reply]
I see :)

**Reply by mwarvik** [reply]
Jeg har noen ganger skrevet "🩸x etter økta", vet ikke om det er vanlig å gjøre.

**Comment by mwarvik** [comment]
Det med å støtte også vanlige runs vil betraktelig øke mengde kall man må gjøre til API-et, men det er vel en ok tradeoff.

**Comment by mwarvik** [comment]
Vanligvis er det vel en ganske stor forskjell på tempo på drag og pauser/opp/nedjogg, så tenker ikke man trenger å stresse veldig mye med clustering-problemet, i alle fall ikke i starten 🤔 (Delte opp i tre kommentarer så man kan diskutere de litt separat)

**Reply by fyksen** [reply]
Jeg er enig.
